test/models.js

```diff
@@ -1,17 +1,104 @@
 var chai = require('chai');
-var should = chai.should();
-var User = require('../src/models/User');
+var assert = chai.assert;
+var User = require('../src/models/user');
+var mockKnex = require('mock-knex');
+const db = require('../lib/db');
+const tracker = mockKnex.getTracker();
 
-describe('User Model', function() {
-  xit('should create a new user', function(done) {
-  });
+describe('User Model', () => { 
+  describe('User Model Creation', () => {
+    let users = [];
+    before(() => {
+      mockKnex.mock(db, 'knex@0.12');
+      tracker.install();
+      tracker.on('query', (query) => {
+        if(query.method === 'insert') {
+          console.log(query);
+          users.push({ username: query.bindings[2], email: query.bindings[0], password: query.bindings[1] })
+        }
+        query.response(users);
+      });
+    });
 
-  xit('should not create a user with the unique email', function(done) {
-  });
+    it('should create a new user', (done) => {
+      User.create({username: "test", email: "test@test.com", password: "test"}).then((user)=>{
+        User.first(['id', 'username', 'email'], {username : "test"}).then((user) => {
+          assert.typeOf(user, 'object');
+          assert.equal(user.username, "test");
+          done();
+        });
+      });
+    });
 
-  xit('should find user by email', function(done) {
+    it('should hash password when creating user', (done) => {
+      User.create({username: "test2", email: "test2@test.com", password: "test2"}).then((user)=>{
+        User.first(['password'], {username : "test2"}).then((user) => {
+          assert.typeOf(user, 'object');
+          assert.notEqual(user.password, "test2");
+          // check for bcrypt password header
+          assert.equal(user.password.slice(0,4), "$2a$")
+          done();
+        });
+      });
+    });
+
+    after(() => {

```

Comment:
If this is the same for all tests, put it at the describe level, in an `afterEach`?